### PR TITLE
Bubble Sort in Dart

### LINKED ISSUE #4297: Bubble sort in dart
## 🚀 Feature
Sort an array of inputs using the Bubble sort technique in dart.

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes

### PR DESCRIPTION
**Pull request by ritvij14**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Code for sorting a list using Bubble Sort technique

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

Repl link :- https://replit.com/@ritvij14/DSA-NeoAlgo#BubbleSort.dart

## Related Issues or Pull Requests
Fixes #4297 
(Write your answer here.)

### CODE REVIEW

**Review by iamrajiv — APPROVED** [review]

File: Dart/sort/BubbleSort.dart [hunk 1]
```diff
@@ -0,0 +1,68 @@
+// function to sort the list
```
Comment (ankitaggarwal23) [inline Dart/sort/BubbleSort.dart]: - Add Multi-Line Comments 
- Add a bit more description

File: Dart/sort/BubbleSort.dart [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  // sorting
+  arr = bubbleSort(arr);
+  print("Sorted list:");
+  for (int i = 0; i < size; i++) {
+    print(arr[i]);
+  }
+}
+/**
+ * Space complexity: O(1)
+ * Time complexity: O(n^2) {worst} / O(n) {best}
```
Comment (ankitaggarwal23) [inline Dart/sort/BubbleSort.dart]: Shift TC and SC after I/P & O/P

File: Dart/sort/BubbleSort.dart [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+ * 1
+ * 2
+ * 3
+ * 4
+ * 5
+ * 6
+ * 7
+ * 8
+ * 9
+ */
```
Comment (ankitaggarwal23) [inline Dart/sort/BubbleSort.dart]: Just a sample test case so use 5 elements is enough for sample testcase and also add a new line at the EOF

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

File: Dart/sort/BubbleSort.dart (hunk 1 above)
Comment (ritvij14) [reply Dart/sort/BubbleSort.dart]: Done.

File: Dart/sort/BubbleSort.dart (hunk 2 above)
Comment (ritvij14) [reply Dart/sort/BubbleSort.dart]: Done

File: Dart/sort/BubbleSort.dart (hunk 3 above)
Comment (ritvij14) [reply Dart/sort/BubbleSort.dart]: @ankitaggarwal23 I have shortened my test case. Also in my code on my local repo, there already was a new line at EOF.

**Comment by ritvij14** [comment]
I have made the necessary changes and also added a repl link for the code incase someone wants to test it out.

File: Dart/sort/BubbleSort.dart [hunk 4]
```diff
@@ -0,0 +1,67 @@
+/**
+Function to sort list using bubble sort technique.
+We use only one extra temporary variable so space
+complexity is O(1). Best case time complexity (if the list
+is already sorted) is O(n).
+ */
```
Comment (ankitaggarwal23) [inline Dart/sort/BubbleSort.dart]: Don't add TC in description remove it and write a beginner-friendly description/ approach

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

File: Dart/sort/BubbleSort.dart (hunk 4 above)
Comment (ritvij14) [reply Dart/sort/BubbleSort.dart]: @ankitaggarwal23 done.

File: Dart/sort/BubbleSort.dart [hunk 5]
```diff
@@ -0,0 +1,66 @@
+/**
+Function to sort list using bubble sort technique.
+We use one extra temporary variable to help us out in swapping. 
+isSorted checks if the list is already sorted.
+ */
+import 'dart:io';
+
+List bubbleSort(List arr) {
```
Comment (ankitaggarwal23) [inline Dart/sort/BubbleSort.dart:9]: Add a one line function description

File: Dart/sort/BubbleSort.dart [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    arr.add(int.parse(stdin.readLineSync()!));
+  }
+  // sorting
+  arr = bubbleSort(arr);
+  print("Sorted list:");
+  for (int i = 0; i < size; i++) {
+    print(arr[i]);
+  }
+}
+/**
```
Comment (ankitaggarwal23) [inline Dart/sort/BubbleSort.dart:47]: Add a new line above /**

File: Dart/sort/BubbleSort.dart [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+ * 0
+ * 1
+ * 2
+ * 3
+ * 4
+ * 5
+ *
+ * Space complexity: O(1)
+ * Time complexity: O(n^2) {worst} / O(n) {best}
+ */
```
Comment (ankitaggarwal23) [inline Dart/sort/BubbleSort.dart:68]: Add a new line at the end of file

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

File: Dart/sort/BubbleSort.dart (hunk 7 above)
Comment (ritvij14) [reply Dart/sort/BubbleSort.dart:68]: image

@ankitaggarwal23 Sir, how do I add it? It's already being shown in VS Code.

File: Dart/sort/BubbleSort.dart (hunk 5 above)
Comment (ritvij14) [reply Dart/sort/BubbleSort.dart:9]: Done

File: Dart/sort/BubbleSort.dart (hunk 6 above)
Comment (ritvij14) [reply Dart/sort/BubbleSort.dart:47]: Done

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**